src/parser/expressions_parser.rs

```diff
@@ -586,6 +578,28 @@ fn parse_literal_number(
     })
 }
 
+fn parse_strict_literal_integer(lexer: &mut ParseSession) -> Result<AstStatement, Diagnostic> {
+    //correct the location if we just parsed a minus before
+    let location = lexer.location();
```

Comment:
Hm? Did you copy this from somewhere?

Reply:
What about the minus?

Reply:
Updated the docs, this method just gets an integer out of the current token, meaning it will not consider a '.' after the token and will also fail if the token contains an exponent.
I didn't consider negatives here because I assumed negatives cannot occur in a bitwise access, but the name of this method impiles it might accept a sign. Do you prefer I add the check or do you have a better expressive name? `parse_natural_number` maybe?